mempool/clist_mempool.go

```diff
@@ -780,3 +784,42 @@ func (rc *recheck) setRecheckFull() bool {
 func (rc *recheck) consideredFull() bool {
 	return rc.recheckFull.Load()
 }
+
+// CListIterator implements an Iterator that traverses the CList sequentially. When the current
+// entry is removed from the mempool, the iterator starts from the beginning of the CList. When it
+// reaches the end, it waits until a new entry is appended.
+type CListIterator struct {
+	txs    *clist.CList    // to wait on and retrieve the first entry
+	cursor *clist.CElement // pointer to the current entry in the list
+}
+
+func (mem *CListMempool) NewIterator() Iterator {
+	return &CListIterator{
+		txs: mem.txs,
+	}
+}
+
+// WaitNextCh returns a channel to wait for the next available entry. The channel will be closed
+// once the entry is added to the channel or when reaching the end of the list.
```

Comment:
Is it safe to call this function from several goroutines? (I don't think it is). Can you clarify this in this comment?

Reply:
It is not and we don't do that. What is safe is to create multiple iterators on the same CList, as we do with one `broadcastTxRoutine` per peer. I'll add a comment to clarify though.